File: csharp/DetectCapital.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode
{
    class DetectCapital
    {
        public bool DetectCapitalUse(string word)
        {
            bool caseAllUpper = true;
            bool caseAllLower = true;
            bool caseTitle = true;
            for(int i = 0; i < word.Length; i++)
            {
                char c = word[i];
                caseAllUpper = caseAllUpper && Char.IsUpper(c);
                caseAllLower = caseAllLower && Char.IsLower(c);
                if (i == 0)
                {
                    caseTitle = caseTitle && Char.IsUpper(c);
                }
                else
                {
                    caseTitle = caseTitle && Char.IsLower(c);
                }
            }

            return caseAllUpper || caseAllLower || caseTitle;
        }
    }
}

File: python/DetectCapital.py



class Solution:
    def detectCapitalUse(self, word):
        case_all_upper = True
        case_all_lower = True
        case_title = True
        for i in range(0, len(word)):
            c = word[i]
            case_all_upper = case_all_upper and c.isupper()
            case_all_lower = case_all_lower and not c.isupper()
            if i == 0:
                case_title = case_title and c.isupper()
            else:
                case_title = case_title and not c.isupper()
        return case_title or case_all_lower or case_all_upper
